rs/bitcoin/adapter/test_utils/src/rpc_json.rs

```diff
@@ -0,0 +1,305 @@
+#[derive(Serialize, Clone, PartialEq, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignRawTransactionInput {
+    pub txid: Txid,
+    pub vout: u32,
+    pub script_pub_key: ScriptBuf,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub redeem_script: Option<ScriptBuf>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "bitcoin::amount::serde::as_btc::opt"
+    )]
+    pub amount: Option<Amount>,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignRawTransactionResultError {
+    pub txid: Txid,
+    pub vout: u32,
+    pub script_sig: ScriptBuf,
+    pub sequence: u32,
+    pub error: String,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignRawTransactionResult {
+    #[serde(with = "serde_hex")]
+    pub hex: Vec<u8>,
+    pub complete: bool,
+    pub errors: Option<Vec<SignRawTransactionResultError>>,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct LoadWalletResult {
+    pub name: String,
+    pub warning: Option<String>,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct UnloadWalletResult {
+    pub warning: Option<String>,
+}
+
+/// deserialize_bip70_network deserializes a Bitcoin Core network according to BIP70
+/// The accepted input variants are: {"main", "test", "signet", "regtest"}
+fn deserialize_bip70_network<'de, D>(deserializer: D) -> Result<Network, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    struct NetworkVisitor;
+    impl<'de> serde::de::Visitor<'de> for NetworkVisitor {
+        type Value = Network;
+
+        fn visit_str<E: serde::de::Error>(self, s: &str) -> Result<Self::Value, E> {
+            Network::from_str(s).map_err(|_| {
+                E::invalid_value(
+                    serde::de::Unexpected::Str(s),
+                    &"bitcoin network encoded as a string",
+                )
+            })
+        }
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            write!(formatter, "bitcoin network encoded as a string")
+        }
+    }
+
+    deserializer.deserialize_str(NetworkVisitor)
+}
+
+/// A module used for serde serialization of bytes in hexadecimal format.
+///
+/// The module is compatible with the serde attribute.
+pub mod serde_hex {
+    pub use bitcoin::hex;
+
+    use hex::{DisplayHex, FromHex};
+    use serde::de::Error;
+    use serde::{Deserializer, Serializer};
+
+    pub fn serialize<S: Serializer>(b: &Vec<u8>, s: S) -> Result<S::Ok, S::Error> {
+        s.serialize_str(&b.to_lower_hex_string())
+    }
+
+    pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<Vec<u8>, D::Error> {
+        let hex_str: String = ::serde::Deserialize::deserialize(d)?;
+        FromHex::from_hex(&hex_str).map_err(D::Error::custom)
+    }
+
+    pub mod opt {
```

Comment:
It seems that this opt serde module is not used in the above type defintions. Could it be removed?